BodoSQL/calcite_sql/bodosql-calcite-application/src/main/java/org/apache/calcite/sql/SqlIntervalQualifier.java

```diff
@@ -461,6 +462,53 @@ public int getIntervalSign(String value) {
     return sign;
   }
 
+  public static TimeUnit stringToDatePartTimeUnit(String stringValue) {
```

Comment:
How is this used? We may need to intercept the usage to ensure we have nanosecond handling.